### PR TITLE
[TRTLLM-13022][test] remove deprecated models from tests

### PR DESCRIPTION
**Pull request by xinhe-nv**
## Summary by CodeRabbit

* **Tests**
  * Removed test coverage for legacy model variants (Llama v2, Gemma 3, EXAONE) and LoRA-related functionality across multiple test suites.
  * Updated test configurations to focus on newer model versions and simplified test matrices.
  * Removed model-specific test fixtures, examples, and benchmark entries.

* **Chores**
  * Consolidated test lists and removed obsolete test cases from CI pipelines.
  * Updated test data files and model path mappings.

[Review Change Stack](https://app.coderabbit.ai/change-stack/NVIDIA/TensorRT-LLM/pull/14660?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

## Description
- delete deprecated tests based on https://docs.google.com/spreadsheets/d/1GGrxEJ6qCmMXgpw_OhFM80N9lcEm3dqci_M-_lrt2RM/edit?gid=1239703108#gid=1239703108

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tests/integration/defs/kv_cache/test_kv_cache_v2_scheduler.py [hunk 1]
```diff
@@ -341,148 +339,6 @@ def test_eviction_overlap(self):
         )
 
 
-# ===========================================================================
-# LoRA tests on llama-7b-hf
-# ===========================================================================
-@pytest.mark.skip_less_device_memory(40000)
-class TestKVCacheV2LoRA:
```
Comment (QiJune) [inline tests/integration/defs/kv_cache/test_kv_cache_v2_scheduler.py:348]: This test is a LoRA feature test. Although llama-7b-hf is used, we should keep it.

Comment (xinhe-nv) [reply tests/integration/defs/kv_cache/test_kv_cache_v2_scheduler.py:348]: this file is restored.

File: tests/integration/defs/test_e2e.py [hunk 2]
```diff
@@ -28,9 +28,7 @@
 from defs.trt_test_alternative import (check_call, check_call_negative_test,
                                        check_output, print_info, print_warning)
 
-from .common import (PluginOptions, convert_weights, get_mmlu_accuracy,
-                     prune_checkpoint, quantize_data, refit_model,
-                     venv_check_call)
+from .common import convert_weights, get_mmlu_accuracy, venv_check_call
```
[the remark this answers is not in the document]
Comment (xinhe-nv) [reply tests/integration/defs/test_e2e.py:30]: fixed

File: tests/unittest/trt/model/test_llama.py [hunk 3]
```diff
@@ -436,28 +436,20 @@ def test_llama(self, use_refit, fast_building, context_fmha_flag,
...
+            ])))
         test_cases.extend(
             list(
-                product([
-                    ("llama-7b-4gqa-hf", "7B-4GQA"),
-                ], [(8, 0), (8, 7)], [-1, 0, 1])))
+                product([], [(1, 0), (2, 0), (2, 1), (4, 0), (4, 1)],
+                        [-1, 0, 1])))
+        test_cases.extend(list(product([], [(8, 0), (8, 7)], [-1, 0, 1])))
         return test_cases
```
[the remark this answers is not in the document]
Comment (xinhe-nv) [reply tests/unittest/trt/model/test_llama.py:453]: fixed

File: tests/integration/defs/accuracy/references/mmlu.yaml [hunk 4]
```diff
@@ -465,11 +450,6 @@ nvidia/NVIDIA-Nemotron-Nano-9B-v2:
   - quant_algo: NVFP4
     kv_cache_quant_algo: FP8
     accuracy: 73.977
-GLM-4.7-Flash:
```
Comment (nvchenghaoz) [inline tests/integration/defs/accuracy/references/mmlu.yaml:468]: @suyoggupta  Do we want to keep the GLM 4.7 Flash for AD accuracy test?

Comment (suyoggupta) [reply tests/integration/defs/accuracy/references/mmlu.yaml:468]: yes. what's the reason to remove it?

Comment (xinhe-nv) [reply tests/integration/defs/accuracy/references/mmlu.yaml:468]: @suyoggupta the requirement is from https://docs.google.com/spreadsheets/d/1GGrxEJ6qCmMXgpw_OhFM80N9lcEm3dqci_M-_lrt2RM/edit?gid=1239703108#gid=1239703108

File: tests/unittest/llmapi/test_llm_multi_gpu_pytorch.py [hunk 5]
```diff
@@ -42,31 +39,6 @@ def test_tinyllama_logits_processor_2gpu(tp_size: int, pp_size: int):
...
-@pytest.mark.gpu2
-def test_llama_7b_lora_tp2():
-    llama_7b_lora_from_dir_test_harness(tensor_parallel_size=2,
-                                        kv_cache_config=global_kv_cache_config)
-
-
-@pytest.mark.gpu4
-@skip_ray  # https://nvbugs/5682551
-@test_lora_with_and_without_cuda_graph
-def test_llama_7b_multi_lora_tp4(cuda_graph_config):
```
Comment (QiJune) [inline tests/unittest/llmapi/test_llm_multi_gpu_pytorch.py:54]: This is a lora feature test, please keep it.

File: tests/unittest/llmapi/test_llm_multi_gpu.py [hunk 6]
```diff
@@ -243,40 +239,6 @@ def test_tinyllama_logits_processor_tp2pp2():
                                             pipeline_parallel_size=2)
 
 
-@pytest.mark.gpu2
-@pytest.mark.part3
-def test_llama_v2_13b_lora_tp2():
```
Comment (QiJune) [inline tests/unittest/llmapi/test_llm_multi_gpu.py:248]: These tests are for lora features, please keep them.

File: tests/unittest/llmapi/test_llm.py [hunk 7]
```diff
@@ -542,41 +535,6 @@ async def main():
     test_non_streaming_usage_wait()
 
 
-@pytest.mark.parametrize("chunked", [True, False])
-@pytest.mark.part0
-@pytest.mark.mpi_ray_parity
-def test_llm_generate_async_with_stream_interval(chunked):
```
Comment (QiJune) [inline tests/unittest/llmapi/test_llm.py:548]: This test is for stream_interval feature, please keep it.

File: tests/unittest/llmapi/test_llm.py [hunk 8]
```diff
@@ -869,93 +827,6 @@ def test_generate_with_stop_words():
                      stop_reasons=["I J"])
 
 
-@force_ampere
-@pytest.mark.part0
-@pytest.mark.parametrize("model_path", [
-    get_model_path('gemma/gemma-3-1b-it'),
-])
-def test_generate_with_detokenization_stop_words(model_path):
```
Comment (QiJune) [inline tests/unittest/llmapi/test_llm.py:877]: This test is for stop words, please keep it.

File: tests/unittest/llmapi/test_llm.py [hunk 9]
```diff
@@ -869,93 +827,6 @@ def test_generate_with_stop_words():
...
-                     finish_reasons=['stop'],
-                     stop_reasons=["there"])
-
-
-@force_ampere
-@pytest.mark.part0
-@pytest.mark.parametrize("model_path", [
-    get_model_path('gemma/gemma-3-1b-it'),
-])
-def test_generate_with_detokenization_stop_words_streaming(model_path):
```
Comment (QiJune) [inline tests/unittest/llmapi/test_llm.py:931]: This test is for stop words, please keep it.

File: tests/unittest/llmapi/test_llm.py [hunk 10]
```diff
@@ -1449,208 +1320,6 @@ def test_executor_lookahead_decoding_config():
     assert sampling_params.lookahead_config.max_verification_set_size == 8
 
 
-def llama_v2_13b_lora_from_dir_test_harness(**llm_kwargs):
```
Comment (QiJune) [inline tests/unittest/llmapi/test_llm.py:1452]: Please keep lora related tests.

File: tests/integration/defs/accuracy/references/mmlu.yaml (hunk 4 above)
Comment (StanleySun639) [reply tests/integration/defs/accuracy/references/mmlu.yaml:468]: @nvchenghaoz @suyoggupta `GLM-4.7-Flash` was reviewed by @laikhtewari  as deprecated model. Laikh, could you comment on this?

**Review by jieli-matrix — APPROVED** [review]

**Review by 2ez4bz — APPROVED** [review]

**Comment by 2ez4bz** [comment]
Thanks for doing this!

File: tests/unittest/_torch/modeling/test_modeling_exaone4.py [hunk 11]
```diff
@@ -1,431 +0,0 @@
-import json
```
Comment (2ez4bz) [inline tests/unittest/_torch/modeling/test_modeling_exaone4.py:1]: cc @yechank-nvidia

Comment (yechank-nvidia) [reply tests/unittest/_torch/modeling/test_modeling_exaone4.py:1]: I'm good on this. EXAONE-4.5 covers EXAONE-4.0.

File: tests/unittest/llmapi/test_executor.py [hunk 12]
```diff
@@ -37,132 +31,6 @@ def engine_path():
     return Path(tempfile.tempdir) / "llm_engine"
 
 
-@pytest.fixture(scope="module")
-def llama_7b_path(engine_path: Path) -> Path:
```
Comment (QiJune) [inline tests/unittest/llmapi/test_executor.py:41]: Please keep all test cases in test_executor.py

File: tests/integration/test_lists/test-db/l0_l40s.yml [hunk 13]
```diff
@@ -52,14 +52,11 @@ l0_l40s:
...
   - unittest/trt/quantization
   - unittest/trt/functional # 37 mins
-  - accuracy/test_cli_flow.py::TestLlama2_7B::test_fp8
   - accuracy/test_cli_flow.py::TestLlama3_1_8B::test_fp8_rowwise_meta_recipe
   - accuracy/test_cli_flow.py::TestGemma2_9BIt::test_auto_dtype
   - accuracy/test_cli_flow.py::TestPhi3Small128kInstruct::test_auto_dtype
   - accuracy/test_llm_api.py::TestLlama3_1_8B::test_fp8_rowwise
-  - examples/test_llama.py::test_llm_llama_v2_lora_1gpu[chinese-llama-2-lora-13b-llama-v2-13b-hf-lora_fp16-base_fp16]
   - examples/test_llama.py::test_llm_llama_v3_dora_1gpu[commonsense-llama-v3-8b-dora-r32-llama-v3-8b-hf-base_fp16]
-  - examples/test_llama.py::test_llm_llama_v1_1gpu_kv_cache_reuse_with_prompt_table[llama-7b]
```
Comment (QiJune) [inline tests/integration/test_lists/test-db/l0_l40s.yml:62]: Please keep this test, which is for kv cache manager

**Review by QiJune — APPROVED** [review]
LGTM

File: tests/integration/test_lists/test-db/l0_l40s.yml (hunk 13 above)
Comment (xinhe-nv) [reply tests/integration/test_lists/test-db/l0_l40s.yml:62]: @QiJune the two cases are trt flow, will we keep them in L0?

**Review by ruodil — APPROVED** [review]
